src/generators/cplusplus/CplusplusConstrainer.ts

```diff
@@ -0,0 +1,71 @@
+import { TypeMapping } from '../../helpers';
+import {
+  defaultEnumKeyConstraints,
+  defaultEnumValueConstraints
+} from './constrainer/EnumConstrainer';
+import { defaultModelNameConstraints } from './constrainer/ModelNameConstrainer';
+import { defaultPropertyKeyConstraints } from './constrainer/PropertyKeyConstrainer';
+import { CplusplusDependencyManager } from './CplusplusDependencyManager';
+import { CplusplusOptions } from './CplusplusGenerator';
+
+export const CplusplusDefaultTypeMapping: TypeMapping<
+  CplusplusOptions,
+  CplusplusDependencyManager
+> = {
+  Object({ constrainedModel }): string {
+    //Returning name here because all object models have been split out
+    return `${constrainedModel.name}*`;
+  },
```

Comment:
IIUC the Object should be different from Reference in that it owns the property / field data, instead of referencing another object. In that case the `*` for pointer type should not be here, and the property's type definition must be visible (by including that type's header).